cookbooks/kinematically_driven_subduction_2d/isotherm_depth.h

```diff
@@ -0,0 +1,74 @@
+/*
+  Copyright (C) 2011-2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__postprocess_isotherm_depth_h
+#define __aspect__postprocess_isotherm_depth_h
+
+#include <aspect/postprocess/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+
+    /**
+     * A postprocessor that computes some statistics about the deepest point of the last composition.
+     *
+     * @ingroup Postprocessing
+     */
+    template <int dim>
+    class IsothermDepth : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Evaluate the solution for some velocity statistics.
```

Comment:
Is this object actually evaluating the solution, or just storing the results of the evaluation? From the .cc code, it looks like it is just used to store the values?